mod.ts

```diff
@@ -8,22 +8,46 @@ import { upgrade } from "./src/commands/upgrade.ts";
 
 import { version } from "./src/version.ts";
 
-import { setupLog } from "./src/log.ts";
+import { handleError, writeLogFile, setupLog } from "./src/log.ts";
 
 await setupLog();
 
-await new Command()
-  .name("eggs")
-  .version(version)
-  .description(
-    "nest.land - A module registry and CDN for Deno, on the permaweb",
-  )
-  .command("help", new HelpCommand())
-  .command("completions", new CompletionsCommand())
-  .command("link", link)
-  .command("init", init)
-  .command("publish", publish)
-  .command("update", update)
-  .command("install", install)
-  .command("upgrade", upgrade)
-  .parse(Deno.args);
+try {
+  const eggs = new Command<Options, Arguments>()
+    .throwErrors()
+    .name("eggs")
+    .version(version)
+    .description(
+      "nest.land - A module registry and CDN for Deno, on the permaweb",
+    )
+    .option("-d, --debug", "Print additional information.", { global: true })
+    .option(
+      "-o, --output-log",
+      "Create a log file after command completion.",
+      { global: true },
+    )
+    .action(() => {
+      eggs.help();
+    })
+    .command("help", new HelpCommand())
+    .command("completions", new CompletionsCommand())
+    .command("link", link)
+    .command("init", init)
+    .command("publish", publish)
+    .command("update", update)
+    .command("install", install)
+    .command("upgrade", upgrade);
+  const { options } = await eggs.parse(Deno.args);
+
+  if (options.outputLog) {
+    await writeLogFile();
+  }
+} catch (err) {
+  await handleError(err);
+}
+
+type Options = {
+  debug: boolean;
+  outputLog: boolean;
+};
+type Arguments = [];
```

Comment:
You're gonna get tired of this but: Move types below imports or above related component

Reply:
I'm gonna change that 👍